Frontend/src/auth/context/firebase/auth-provider.tsx

```diff
@@ -0,0 +1,214 @@
+import { useMemo, useEffect, useReducer, useCallback } from 'react';
+import {
+  getAuth,
+  signOut,
+  OAuthProvider,
+  signInWithPopup,
+  onAuthStateChanged,
+  GoogleAuthProvider,
+  FacebookAuthProvider,
+} from 'firebase/auth';
+
+// import axios, { endpoints } from 'src/utils/axios';
+
+import { AuthContext } from './auth-context';
+import { firebaseApp, isRegistered as isRegister } from './lib';
+import { AuthUserType, ActionMapType, AuthStateType } from '../../types';
+// ----------------------------------------------------------------------
+
+const AUTH = getAuth(firebaseApp);
+
+enum Types {
+  INITIAL = 'INITIAL',
+  REGISTER = 'REGISTER',
+}
+
+type Payload = {
+  [Types.INITIAL]: {
+    user: AuthUserType;
```

Comment:
same thing, firebase user is not the same as full user object. name them as such. jwtUser or Firebase User 

and the other one 
just user

Reply:
Ok :)

Reply:
OK :>